apps/acrs/behaviours/resume-form-session.js

```diff
@@ -0,0 +1,203 @@
+
+        if (idType === 'brp') {
+          isSameCase = _.get(cases, "[0].session['brp']") === brp;
+          idNumber = brp;
+        }else{
+          isSameCase = _.get(cases, "[0].session['uan']") === uan;
+          idNumber = uan;
+        }
+
+
+        const singleCase = cases.length < 2;
+
+        const noCaseOrSameCase = !cases[0] || isSameCase;
+        const multipleCasesInSession = _.get(req.sessionModel.get('user-cases'), 'length') > 1;
+
+        if (singleCase && noCaseOrSameCase && !multipleCasesInSession) {
+          if (cases[0]) {
+            this.setupSession(req, cases[0].session);
+          }
+          req.sessionModel.set('multipleCases', false);
+          return res.redirect(super.getNextStep(req, res, next));
+        }
+
+        const filteredCases = this.addCasesToSession(req, cases, idNumber);
+        this.setupRadioButtons(req, filteredCases.filter(form => form.id));
+
+        return super.configure(req, res, next);
+      })
+      .catch(next);
+  }
+
+  addCasesToSession(req, cases, idNumber) {
+    const idType = req.sessionModel.get('id-type');
+    let caseAlreadyInSession = '';
+
+    if (idType === 'brp') {
+      caseAlreadyInSession = cases.find(obj => obj.session.brp === idNumber);
+    }else{
+      caseAlreadyInSession = cases.find(obj => obj.session.uan === idNumber);
+    }
+
+    if (!caseAlreadyInSession) {
+      this.addSessionCaseToList(req, cases);
+    }
+
+    const filteredCases = cases.filter(caseType => {
+      const isDuplicate = cases.filter(obj => {
+        if (idType === 'brp') {
+          return obj.session.brp === caseType.session.brp;
+        }
+        if (idType === 'uan') {
+          return obj.session.uan === caseType.session.uan;
+        }
+      }).length > 1;
+
+      return caseType.id || (!caseType.id && !isDuplicate);
+    });
+
+    req.sessionModel.set('user-cases', filteredCases);
+    return filteredCases;
+  }
+
+  addSessionCaseToList(req, cases) {
+    const defaultSession = {
+      session: Object.assign(...SESSION_DEFAULTS.fields.map(field => {
+        return { [field]: req.sessionModel.get(field) };
+      }))
+    };
+    defaultSession.session.steps = SESSION_DEFAULTS.steps;
+    cases.push(defaultSession);
+  }
+
+  parseCasesSessions(body) {
+    let cases = body;
+
+    if (cases.length) {
+      cases = cases.map(caseType => {
+        const session = caseType.session;
+        caseType.session = typeof session === 'string' ?
+          JSON.parse(session) : session;
+
+        return caseType;
+      });
+    }
+    return cases;
+  }
+  // POST lifecycle
+  saveValues(req, res, next) {
+    const cases = req.sessionModel.get('user-cases') || [];
+    const idType = req.sessionModel.get('id-type');
+    let caseObj = [];
+
+    if (idType === 'brp') {
+      caseObj = cases.find(obj => obj.session.brp === req.form.values['id-type']);
+    }else{
+      caseObj = cases.find(obj => obj.session.uan === req.form.values['id-type']);
+    }
```

Comment:
same as above you could do `const selectedCase = cases.find(obj => obj.session[idType] === req.form.values['id-type']`
then use `selectedCase` for further setup

Reply:
but this will work ok too. However, I still think 'id-type' is a confusing form field name given we also have `idType` and `sessionModel['id-type']`

Reply:
Declaring a const make it immutable. idType is only declared in the block while id-type is for session.

Reply:
Don't worry about this change if you want - just saying you can make the whole block a one-liner if you like.

Reply:
do you mean do it this way ?

const casObj = (idType === 'brp') ? cases.find(obj => obj.session.brp === req.form.values['id-type']):caseObj = cases.find(obj => obj.session.uan === req.form.values['id-type']);

Reply:
No I mean that `idType` is either 'brp' or 'uan' so if you do `obj.session[idType]` it will get `obj.session.uan` or `obj.session.brp` depending on what the value of `idType` was. Just means you don't need an if else block to make the conditional.

Reply:
but if you like the if else block more then keep it - I have no problem with it 🙂

Reply:
done